app/src/main/java/org/stepik/android/presentation/course/CoursePresenter.kt

```diff
@@ -167,13 +174,7 @@ constructor(
             ?.takeIf { it.stats.enrollmentState != EnrollmentState.Pending }
             ?: return
 
-        state = CourseView.State.BlockingLoading(
-            headerData.copy(
-                stats = headerData.stats.copy(
-                    enrollmentState = EnrollmentState.Pending
-                )
-            )
-        )
+        view?.setBlockingLoading(isLoading = true)
```

Comment:
Is loading = false is never set. We can leave pending status for enrolment updates. And use blocking loading only for continue learning.